[Feature] LogKit 생성 및 로그 출력 방식 일괄 통일

NeoSelf1 opened the pull request:
## ✅ 작업 사항
- LogKit 패키지 새로 구현
- BookMatchLogger, NetworkEventLogger, DSLogger 등 기타 패키지에서 별도로 생성되어 호출되던 OSLog 인스턴스 일괄 제거
- LogKit import하여 LogKit 패키지의 출력 메서드로 일괄 통일
- SubSystem, Category에 대한 열거형 정의 및 메서드에 사용
- 간편한 호출방식을 위한 Facade 계층 새로 추가

## 👉 리뷰 포인트
![스크린샷 2025-03-02 오후 2 16 33](https://github.com/user-attachments/assets/39747641-2ec6-4e61-89d9-6d5ec0539c24)
![스크린샷 2025-03-02 오후 2 16 26](https://github.com/user-attachments/assets/085c0c3a-76c5-48f4-a274-fb28172befc5)
![스크린샷 2025-03-02 오후 2 16 19](https://github.com/user-attachments/assets/a0da84a7-6266-4294-9971-2803a1238399)
![스크린샷 2025-03-02 오후 2 16 01](https://github.com/user-attachments/assets/2075eaf6-9424-498c-9e1b-3238c02b8330)
![스크린샷 2025-03-02 오후 2 18 15](https://github.com/user-attachments/assets/9799e964-341b-4e41-bb45-c657b375d671)

NeoSelf1 committed:
[Feature] DiskCache 클래스 구현 (#109)

NeoSelf1 committed:
[Delete] 불필요 라이브러리 import문 제거 (#109)

NeoSelf1 committed:
[Refactor] 동시성 제어 관련 컴파일 에러 해결 (#109)

- MemoryStorage -> MemoryStorageActor로 변경하여 내부 메서드가 직렬화 실행될 수 있도록 보장
- NSLock 객체 제거

NeoSelf1 committed:
[Refactor] Actor 어노테이션 불필요한 곳 수정 (#109)

NeoSelf1 committed:
[Feat] ImageProcessor 추가 (#109)

NeoSelf1 committed:
[Feat] ImageWrapper 추가 (#109)

NeoSelf1 committed:
[Feat] ImageDownloader 구현 (#109)

NeoSelf1 committed:
[Fix] Task-isolated options 관련 에러 수정 위해 임시 커밋 (#109)

NeoSelf1 committed:
[Fix] Concurrency 문제 해결 (#109)

NeoSelf1 committed:
[Fix] NeoImageWrapper 자체를 Sendable하게 변경하여 해결 (#109)

NeoSelf1 committed:
[Refactor] ImageCache 패키지 구현 (#109)

NeoSelf1 committed:
[Refactor] 디렉토리 구조 변경 (#109)

NeoSelf1 committed:
[Feat] 메모리 및 디스크 저장소 조회하여 기존 이미지 가져오는 로직 추가 (#109)

NeoSelf1 committed:
[Refactor] BookMatchKit loggers 초기과 구문 임시 저장

NeoSelf1 committed:
[Feat] LogCategory, LogSubsystem, LogLevel 열거형 정의

NeoSelf1 committed:
[Feat] LogKitActor 및 LogKit 클래스 구현체 구현

NeoSelf1 committed:
[Feat] LogKit Facade 계층 구현

- 복잡한 내부 동작 감추고, 단순한 외부 인터페이스만을 제공

NeoSelf1 committed:
[Refactor] CSV 파일 형식에 맞춰 다중라인 출력 String문 단일라인으로 수정

NeoSelf1 committed:
[Delete] 각종 모듈에서 사용되고 있던 Logger 일괄 제거 및 LogKit로 통합

NeoSelf1 committed:
[Refactor] 각 패키지에 대해 LogKit 패키지 의존성 추가

NeoSelf1 committed:
[Refactor] Log 관련 메서드 모두 LogKit 패키지 메서드로 일괄 변경

NeoSelf1 committed:
[Refactor] DesignSystem 라이브러리 내부 Kingfisher 라이브러리 의존성 제거 및 LogKit 내부 gitignore 파일 추가

ericKwon95 commented on BookKitty/BookKitty/LogKit/Sources/LogKit/LogKitActor.swift:
```diff
@@ -0,0 +1,162 @@
+    /// static 프로퍼티 및 메서드는 인스턴스와 무관하게 타입 자체에 속하기에 actor의 격리 메커니즘 밖에 존재하여, actor-isolation 제약을 받지 않게
+    /// 됨.
+    private static let loggers: [LoggerKey: Logger] = {
+        var map: [LoggerKey: Logger] = [:]
```
굉장히 사소한 부분이긴 한데요, map 대신 더 구체적인 변수명을 사용하면 좋지 않을까 생각이 들었습니다!

ericKwon95 commented on BookKitty/BookKitty/LogKit/Sources/LogKit/LogKitActor.swift:
```diff
@@ -0,0 +1,162 @@
+    /// 각 카테고리에 해당한느 로거 인스턴스를 Actor 속성이 아닌, 정적 속성으로 생성
+    /// static 프로퍼티 및 메서드는 인스턴스와 무관하게 타입 자체에 속하기에 actor의 격리 메커니즘 밖에 존재하여, actor-isolation 제약을 받지 않게
+    /// 됨.
+    private static let loggers: [LoggerKey: Logger] = {
```
static 변수는 lazy하게 초기화되며 스레드 안전하지 않기 때문에, loggers를 여러 곳에서 동시적으로 호출하지는 않는지 확인이 필요해 보입니다~!

ericKwon95 commented on BookKitty/BookKitty/LogKit/Sources/LogKit/LogKitActor.swift:
```diff
@@ -0,0 +1,162 @@
+    public static let shared = LogKitActor()
+}
+
+public final class _LogKit {
```
_ 를 쓴 이유가 궁금해요!

ericKwon95 commented on BookKitty/BookKitty/LogKit/Sources/LogKit/LogKitActor.swift:
```diff
@@ -0,0 +1,162 @@
+    
+    // MARK: - Public Methods
+    
+    func log(
```
public 선언이 안 되어 있는 듯 합니당

ericKwon95 replied:
인터널로 사용할 계획이셨군요~ 짱짱입니다!

ericKwon95 commented on BookKitty/BookKitty/LogKit/Sources/LogKit/LogKitActor.swift:
```diff
@@ -6,74 +6,80 @@ struct LoggerKey: Hashable {
-/// 단순히 전역 액터 속성만 정의
+/// 단순히 전역 액터 속성만 정의하는 것이 아니라, 로깅 인터페이스도 제공
 @globalActor
 public actor LogKitActor {
```
FileManager 자체가 thread-safe한 것으로 알고 있는데, actor로 한 번 더 감싸주신 이유가 궁금합니다~!

ericKwon95 commented on BookKitty/BookKitty/LogKit/Sources/LogKit/LogKit.swift:
```diff
@@ -0,0 +1,81 @@
+public enum LogKit {
```
퍼사드 패턴에 대해 알아갑니다! 감사합니다~!

ericKwon95 reviewed (approved):
로그가 꽤 많았을텐데,, 전부 고치느라 고생 많으셨습니다!!

ericKwon95 committed:
[Feat] Swift 6 점진적 대응 위한 컴파일 버전 낮추기

ericKwon95 committed:
[Feat] FIFO 순서의 로깅 위한 시리얼큐 적용

- LogEngine으로 이름 변경
- 주석 작성
- 싱글턴 인스턴스가 사용되기 전에는 초기화되지 않기 때문에 앱 시작 시간이라는 보장이 없어 logStartTime으로 네이밍 변경

ericKwon95 committed:
[Refactor] 파일 쓰기, 로그 엔진, 로그 엔진 래퍼로 역할 및 파일 분리
